src/components/SideNavigation/SideNavigation.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { PropsWithChildren, ReactNode } from "react";
+import type { PropsWithSpread } from "types";
+import classNames from "classnames";
+import { type HTMLProps } from "react";
+
+import type { SideNavigationItemProps } from "./SideNavigationItem";
+import SideNavigationItem from "./SideNavigationItem";
+import type {
+  SideNavigationLinkDefaultElement,
+  SideNavigationLinkProps,
+} from "./SideNavigationLink";
+import { isReactNode } from "utils";
+
+export type NavItem<L = SideNavigationLinkDefaultElement> =
+  | SideNavigationItemProps<L>
+  | ReactNode
+  | null;
+
+export type NavItemGroup<L = SideNavigationLinkDefaultElement> = NavItem<L>[];
+
+export type NavGroup<L = SideNavigationLinkDefaultElement> = PropsWithSpread<
+  {
+    /**
+     * The navigation items.
+     */
+    items: NavItemGroup<L>;
+  },
+  HTMLProps<HTMLUListElement>
+>;
+
+export type Props<L = SideNavigationLinkDefaultElement> = PropsWithSpread<
+  {
+    /**
+     * The navigation content. This can be used instead of supplying `items`.
+     */
+    children?: PropsWithChildren["children"];
+    /**
+     * Whether to use the dark theme.
+     */
+    dark?: boolean;
+    /**
+     * Whether the navigation items contain icons.
+     */
+    hasIcons?: boolean;
+    /**
+     * The navigation items. This can be a single array of items, nested arrays
+     * for multiple navigation lists or an object containing items and props
+     * that are passed to the list element.
+     */
+    items?:
+      | NavItemGroup<L>
+      | (NavItemGroup<L> | null)[]
+      | (NavGroup<L> | null)[];
+    /**
+     * The component or element to use for the link elements e.g. `a` or `NavLink`.
+     * @default a
+     */
+    linkComponent?: SideNavigationLinkProps["component"];
+    /**
+     * Classes to apply to the navigation list(s).
+     */
+    listClassName?: string;
+    /**
+     * Classes to apply to the nav element.
+     */
+    navClassName?: string;
+  },
+  HTMLProps<HTMLDivElement>
+>;
+
+const generateItem = <L = SideNavigationLinkDefaultElement,>(
+  item: NavItem<L>,
+  index: number,
+  linkComponent?: Props<L>["linkComponent"],
+  dark?: boolean
+) => {
+  if (isReactNode(item)) {
+    return <SideNavigationItem key={index}>{item}</SideNavigationItem>;
+  }
+  if ("nonInteractive" in item) {
+    return (
+      <SideNavigationItem {...item} dark={item.dark ?? dark} key={index} />
```

Comment:
Can we rely on the parent for the `dark` property? I wonder if the item itself needs a notion of being dark or not -- maybe it can fully rely on its parent for this?
I can't think of a use case where some items should be dark and others or the parent not.

Reply:
Yeah I'm not sure when this might be required, but I figured it can't hurt to override the property if someone specifically sets the property some items.